### ISSUE TITLE
[win10][1.0.4] stacktrace occurs

### ISSUE THREAD

**Issue by kometchtech** [Foundry376/Mailspring#148]
##### Are there any related issues?

...no

##### What operating system are you using?
...Windows 10 Pro 1703

##### What version of Mailspring are you using?
...1.0.4

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?
...no

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
...Gmail / other IMAP

##### Is the issue reproducible with a particular attachment, message, signature, etc?

...
```
13948 [2017-10-13 18:53:24.962] [main] [info] ------------- Starting Sync ---------------
13948 [2017-10-13 18:53:24.985] [background] [info] Fetching XOAuth2 access token for b163048a
13948 [2017-10-13 18:53:24.985] [metadata] [info] Metadata delta stream starting...
13948 [2017-10-13 18:53:25.844] [background] [info] Marking all folders as `busy`
13948 [2017-10-13 18:53:25.846] [background] [info] Syncing folder list...
13948 [2017-10-13 18:53:28.915] [background] [info] Syncing folder list...
13948 [2017-10-13 18:53:29.480] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

14520 [2017-10-13 18:53:29.559] [main] [info] ------------- Starting Sync ---------------
14520 [2017-10-13 18:53:29.570] [background] [info] Fetching XOAuth2 access token for b163048a
14520 [2017-10-13 18:53:29.571] [metadata] [info] Metadata delta stream starting...
14520 [2017-10-13 18:53:30.403] [background] [info] Marking all folders as `busy`
14520 [2017-10-13 18:53:30.405] [background] [info] Syncing folder list...
14520 [2017-10-13 18:53:33.126] [background] [info] Syncing folder list...
14520 [2017-10-13 18:53:33.642] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

1400 [2017-10-13 18:53:33.695] [main] [info] ------------- Starting Sync ---------------
1400 [2017-10-13 18:53:33.701] [background] [info] Fetching XOAuth2 access token for b163048a
1400 [2017-10-13 18:53:33.702] [metadata] [info] Metadata delta stream starting...
1400 [2017-10-13 18:53:34.490] [background] [info] Marking all folders as `busy`
1400 [2017-10-13 18:53:34.678] [background] [info] Syncing folder list...
1400 [2017-10-13 18:53:37.079] [background] [info] Syncing folder list...
1400 [2017-10-13 18:53:37.599] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

8796 [2017-10-13 18:53:37.650] [main] [info] ------------- Starting Sync ---------------
8796 [2017-10-13 18:53:37.657] [background] [info] Fetching XOAuth2 access token for b163048a
8796 [2017-10-13 18:53:37.657] [metadata] [info] Metadata delta stream starting...
8796 [2017-10-13 18:53:38.451] [background] [info] Marking all folders as `busy`
8796 [2017-10-13 18:53:38.452] [background] [info] Syncing folder list...
8796 [2017-10-13 18:53:40.936] [background] [info] Syncing folder list...
8796 [2017-10-13 18:53:44.303] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

8608 [2017-10-13 18:53:44.639] [main] [info] ------------- Starting Sync ---------------
8608 [2017-10-13 18:53:44.650] [background] [info] Fetching XOAuth2 access token for b163048a
8608 [2017-10-13 18:53:44.651] [metadata] [info] Metadata delta stream starting...
8608 [2017-10-13 18:53:45.371] [background] [info] Marking all folders as `busy`
8608 [2017-10-13 18:53:45.373] [background] [info] Syncing folder list...
8608 [2017-10-13 18:53:48.096] [background] [info] Syncing folder list...
8608 [2017-10-13 18:53:50.256] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

9900 [2017-10-13 18:55:44.199] [main] [info] ------------- Starting Sync ---------------
9900 [2017-10-13 18:55:44.208] [background] [info] Fetching XOAuth2 access token for b163048a
9900 [2017-10-13 18:55:44.208] [metadata] [info] Metadata delta stream starting...
9900 [2017-10-13 18:55:44.948] [background] [info] Marking all folders as `busy`
9900 [2017-10-13 18:55:44.949] [background] [info] Syncing folder list...
9900 [2017-10-13 18:55:47.903] [background] [info] Syncing folder list...
9900 [2017-10-13 18:55:50.004] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

1672 [2017-10-13 18:55:50.057] [main] [info] ------------- Starting Sync ---------------
1672 [2017-10-13 18:55:50.064] [background] [info] Fetching XOAuth2 access token for b163048a
1672 [2017-10-13 18:55:50.065] [metadata] [info] Metadata delta stream starting...
1672 [2017-10-13 18:55:50.871] [background] [info] Marking all folders as `busy`
1672 [2017-10-13 18:55:50.872] [background] [info] Syncing folder list...
1672 [2017-10-13 18:55:53.435] [background] [info] Syncing folder list...
1672 [2017-10-13 18:55:55.431] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

8272 [2017-10-13 18:55:55.495] [main] [info] ------------- Starting Sync ---------------
8272 [2017-10-13 18:55:55.505] [background] [info] Fetching XOAuth2 access token for b163048a
8272 [2017-10-13 18:55:55.506] [metadata] [info] Metadata delta stream starting...
8272 [2017-10-13 18:55:56.273] [background] [info] Marking all folders as `busy`
8272 [2017-10-13 18:55:56.274] [background] [info] Syncing folder list...
8272 [2017-10-13 18:55:58.827] [background] [info] Syncing folder list...
8272 [2017-10-13 18:56:00.801] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

8164 [2017-10-13 18:56:00.849] [main] [info] ------------- Starting Sync ---------------
8164 [2017-10-13 18:56:00.856] [background] [info] Fetching XOAuth2 access token for b163048a
8164 [2017-10-13 18:56:00.856] [metadata] [info] Metadata delta stream starting...
8164 [2017-10-13 18:56:01.660] [background] [info] Marking all folders as `busy`
8164 [2017-10-13 18:56:01.661] [background] [info] Syncing folder list...
8164 [2017-10-13 18:56:04.613] [background] [info] Syncing folder list...
8164 [2017-10-13 18:56:06.591] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

13868 [2017-10-13 18:56:06.646] [main] [info] ------------- Starting Sync ---------------
13868 [2017-10-13 18:56:06.657] [background] [info] Fetching XOAuth2 access token for b163048a
13868 [2017-10-13 18:56:06.657] [metadata] [info] Metadata delta stream starting...
13868 [2017-10-13 18:56:07.402] [background] [info] Marking all folders as `busy`
13868 [2017-10-13 18:56:09.749] [main] [info] Marking all folders as `busy`
14876 [2017-10-13 18:56:12.834] [main] [info] ------------- Starting Sync ---------------
14876 [2017-10-13 18:56:12.840] [background] [info] Fetching XOAuth2 access token for b163048a
14876 [2017-10-13 18:56:12.841] [metadata] [info] Metadata delta stream starting...
14876 [2017-10-13 18:56:13.685] [background] [info] Marking all folders as `busy`
14876 [2017-10-13 18:56:13.686] [background] [info] Syncing folder list...
14876 [2017-10-13 18:56:16.085] [background] [info] Syncing folder list...
14876 [2017-10-13 18:56:18.148] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

7780 [2017-10-13 18:56:18.194] [main] [info] ------------- Starting Sync ---------------
7780 [2017-10-13 18:56:18.200] [background] [info] Fetching XOAuth2 access token for b163048a
7780 [2017-10-13 18:56:18.200] [metadata] [info] Metadata delta stream starting...
7780 [2017-10-13 18:56:19.059] [background] [info] Marking all folders as `busy`
7780 [2017-10-13 18:56:19.060] [background] [info] Syncing folder list...
7780 [2017-10-13 18:56:21.441] [background] [info] Syncing folder list...
7780 [2017-10-13 18:56:23.431] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

2196 [2017-10-13 18:56:23.477] [main] [info] ------------- Starting Sync ---------------
2196 [2017-10-13 18:56:23.484] [background] [info] Fetching XOAuth2 access token for b163048a
2196 [2017-10-13 18:56:23.484] [metadata] [info] Metadata delta stream starting...
2196 [2017-10-13 18:56:24.284] [background] [info] Marking all folders as `busy`
2196 [2017-10-13 18:56:24.285] [background] [info] Syncing folder list...
2196 [2017-10-13 18:56:26.679] [background] [info] Syncing folder list...
11476 [2017-10-13 18:56:33.720] [main] [info] ------------- Starting Sync ---------------
11476 [2017-10-13 18:56:33.734] [background] [info] Fetching XOAuth2 access token for b163048a
11476 [2017-10-13 18:56:33.734] [metadata] [info] Metadata delta stream starting...
11476 [2017-10-13 18:56:34.488] [background] [info] Marking all folders as `busy`
11476 [2017-10-13 18:56:34.489] [background] [info] Syncing folder list...
11476 [2017-10-13 18:56:37.050] [background] [info] Syncing folder list...
11476 [2017-10-13 18:56:39.128] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

1052 [2017-10-13 18:56:39.176] [main] [info] ------------- Starting Sync ---------------
1052 [2017-10-13 18:56:39.183] [background] [info] Fetching XOAuth2 access token for b163048a
1052 [2017-10-13 18:56:39.183] [metadata] [info] Metadata delta stream starting...
1052 [2017-10-13 18:56:39.996] [background] [info] Marking all folders as `busy`
1052 [2017-10-13 18:56:39.997] [background] [info] Syncing folder list...
1052 [2017-10-13 18:56:42.500] [background] [info] Syncing folder list...
1052 [2017-10-13 18:56:44.527] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

15036 [2017-10-13 18:56:44.586] [main] [info] ------------- Starting Sync ---------------
15036 [2017-10-13 18:56:44.594] [background] [info] Fetching XOAuth2 access token for b163048a
15036 [2017-10-13 18:56:44.595] [metadata] [info] Metadata delta stream starting...
15036 [2017-10-13 18:56:45.374] [background] [info] Marking all folders as `busy`
15036 [2017-10-13 18:56:55.515] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***

11112 [2017-10-13 18:56:59.386] [main] [info] ------------- Starting Sync ---------------
11112 [2017-10-13 18:56:59.393] [background] [info] Fetching XOAuth2 access token for b163048a
11112 [2017-10-13 18:56:59.394] [metadata] [info] Metadata delta stream starting...
11112 [2017-10-13 18:57:00.217] [background] [info] Marking all folders as `busy`
11112 [2017-10-13 18:57:10.413] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***

11252 [2017-10-13 18:57:10.492] [main] [info] ------------- Starting Sync ---------------
11252 [2017-10-13 18:57:10.502] [background] [info] Fetching XOAuth2 access token for b163048a
11252 [2017-10-13 18:57:10.503] [metadata] [info] Metadata delta stream starting...
11252 [2017-10-13 18:57:11.260] [background] [info] Marking all folders as `busy`
11252 [2017-10-13 18:57:21.398] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***

13916 [2017-10-13 18:57:56.596] [main] [info] ------------- Starting Sync ---------------
13916 [2017-10-13 18:57:56.605] [background] [info] Fetching XOAuth2 access token for b163048a
13916 [2017-10-13 18:57:56.605] [metadata] [info] Metadata delta stream starting...
13916 [2017-10-13 18:57:57.508] [background] [info] Marking all folders as `busy`
13916 [2017-10-13 18:57:57.511] [background] [info] Syncing folder list...
13916 [2017-10-13 18:57:59.961] [background] [info] Syncing folder list...
13916 [2017-10-13 18:58:02.031] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

13216 [2017-10-13 18:58:02.090] [main] [info] ------------- Starting Sync ---------------
13216 [2017-10-13 18:58:02.098] [background] [info] Fetching XOAuth2 access token for b163048a
13216 [2017-10-13 18:58:02.099] [metadata] [info] Metadata delta stream starting...
13216 [2017-10-13 18:58:02.866] [background] [info] Marking all folders as `busy`
13216 [2017-10-13 18:58:02.868] [background] [info] Syncing folder list...
13216 [2017-10-13 18:58:05.430] [background] [info] Syncing folder list...
13216 [2017-10-13 18:58:07.420] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

3412 [2017-10-13 18:58:07.484] [main] [info] ------------- Starting Sync ---------------
3412 [2017-10-13 18:58:07.492] [background] [info] Fetching XOAuth2 access token for b163048a
3412 [2017-10-13 18:58:07.492] [metadata] [info] Metadata delta stream starting...
3412 [2017-10-13 18:58:09.839] [background] [info] Marking all folders as `busy`
3412 [2017-10-13 18:58:09.840] [background] [info] Syncing folder list...
3412 [2017-10-13 18:58:12.350] [background] [info] Syncing folder list...
3412 [2017-10-13 18:58:14.457] [background] [critical]  *** Stack trace:
 ***   raise
 ***   abort
 ***   terminate
 ***   wcsstr
 ***   wcsstr
 ***   NtContinueCHPE
 ***   NtContinueCHPE
 ***   KiUserExceptionDispatcher
 ***   _CxxThrowException
 ***   _set_app_type
 ***   RtlGetAppContainerNamedObjectPath
 ***   RtlGetAppContainerNamedObjectPath
 ***

580 [2017-10-13 18:58:17.481] [main] [info] ------------- Starting Sync ---------------
580 [2017-10-13 18:58:17.491] [background] [info] Fetching XOAuth2 access token for b163048a
580 [2017-10-13 18:58:17.491] [metadata] [info] Metadata delta stream starting...
580 [2017-10-13 18:58:18.326] [background] [info] Marking all folders as `busy`
580 [2017-10-13 18:58:18.327] [background] [info] Syncing folder list...
580 [2017-10-13 18:58:21.521] [background] [info] Syncing folder list...
```
Even after reloading, restarting mailspring did not change the situation.
As a precaution, should I delete Cache and try resetting?

--

**Feature Request?**
##### Does this feature exist in another mail client or tool you use?
...

**Comment by kometchtech** [comment]
In the meantime, prioritizing recovery, deleting Email cache and restarting, it is now reconnected.
stack trace has not occurred.

**Comment by bengotow** [comment]
Hey! Looking at this again, I think this was fixed in 1.0.7 or 1.0.8.
